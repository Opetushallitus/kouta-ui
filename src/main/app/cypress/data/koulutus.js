import { merge } from 'lodash';

const getBaseFields = () => {
  return {
    oid: '1.2.246.562.13.00000000000000000072',
    koulutustyyppi: 'amm',
    koulutusKoodiUri: 'koulutus_0#1',
    tila: 'tallennettu',
    tarjoajat: ['1.2.246.562.10.81934895871'],
    nimi: {
      en: 'Vocational qualification in Agriculture',
      fi: 'Maatalousalan perustutkinto',
      sv: 'Grundexamen inom lantbruksbranschen',
    },
    metadata: {
      tyyppi: 'amm',
      kuvaus: {},
      lisatiedot: [
        {
          otsikkoKoodiUri: 'koulutuksenlisatiedot_0#1',
          teksti: {
            fi: 'koulutuksenlisatiedot_0 kuvaus',
          },
        },
      ],
    },
    julkinen: true,
    muokkaaja: '1.2.246.562.24.62301161440',
    organisaatioOid: '1.2.246.562.10.594252633210',
    kielivalinta: ['fi', 'sv'],
    modified: '2019-04-01T13:01',
  };
};

const getAmmatillinenFields = ({ tyyppi }) => {
  return merge(getBaseFields(), {
    koulutustyyppi: tyyppi,
    koulutusKoodiUri: 'koulutus_0#1',
    tila: 'tallennettu',
    nimi: {
      en: 'Vocational qualification in Agriculture',
      fi: 'Maatalousalan perustutkinto',
      sv: 'Grundexamen inom lantbruksbranschen',
    },
    metadata: { tyyppi },
  });
};

const getKorkeakouluFields = ({ tyyppi }) => {
  return merge(getBaseFields(), {
    johtaaTutkintoon: true,
    koulutustyyppi: tyyppi,
    koulutusKoodiUri: 'koulutus_0#1',
    nimi: { fi: 'Fi nimi', sv: 'Sv nimi' },
    metadata: {
      tyyppi,
      kuvaus: { fi: 'Fi kuvaus', sv: 'Sv kuvaus' },
      tutkintonimikeKoodiUrit: ['tutkintonimikekk_1#1', 'tutkintonimikekk_2#1'],
      opintojenLaajuusKoodiUri: 'opintojenlaajuus_1#1',
      kuvauksenNimi: { fi: 'Fi kuvauksen nimi', sv: 'Sv kuvauksen nimi' },
      koulutusalaKoodiUrit: [
        'kansallinenkoulutusluokitus2016koulutusalataso2_1#1',
      ],
    },
  });
};

const getLukioFields = ({ tyyppi }) => getAmmatillinenFields({ tyyppi });

export default ({ tyyppi = 'amm' } = {}) => {
  if (tyyppi === 'amm') {
    return getAmmatillinenFields({ tyyppi });
  } else if (['yo', 'amk'].includes(tyyppi)) {
    return getKorkeakouluFields({ tyyppi });
  } else if (tyyppi === 'lk') {
    return getLukioFields({ tyyppi });
  }

  return getBaseFields();
};
