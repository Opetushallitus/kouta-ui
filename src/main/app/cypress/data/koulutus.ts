import { merge } from 'lodash/fp';

const getBaseFields = ({ tyyppi = 'amm' } = {}) => {
  return {
    oid: '1.2.246.562.13.00000000000000000072',
    koulutustyyppi: tyyppi,
    koulutuksetKoodiUri: ['koulutus_0#1'],
    tila: 'tallennettu',
    tarjoajat: ['1.2.246.562.10.81934895871'],
    nimi: {
      en: 'Vocational qualification in Agriculture',
      fi: 'Maatalousalan perustutkinto',
      sv: 'Grundexamen inom lantbruksbranschen',
    },
    metadata: {
      tyyppi: tyyppi,
      kuvaus: {},
      lisatiedot: [
        {
          otsikkoKoodiUri: 'koulutuksenlisatiedot_0#1',
          teksti: {
            fi: 'koulutuksenlisatiedot_0 kuvaus',
          },
        },
      ],
    },
    julkinen: true,
    muokkaaja: '1.2.246.562.24.62301161440',
    organisaatioOid: '1.2.246.562.10.594252633210',
    kielivalinta: ['fi', 'sv'],
    modified: '2019-04-01T13:01',
  };
};

const getAmmatillinenFields = ({ tyyppi = 'amm' }) => {
  return merge(getBaseFields(), {
    oid: '1.2.246.562.13.00000000000000000599',
    johtaaTutkintoon: false,
    koulutustyyppi: tyyppi,
    koulutuksetKoodiUri: ['koulutus_351107#12'],
    tila: 'tallennettu',
    tarjoajat: [],
    nimi: {
      en: 'Vocational qualification in Mining',
      fi: 'Kaivosalan perustutkinto',
      sv: 'Grundexamen inom gruvbranschen',
    },
    metadata: {
      tyyppi: tyyppi,
      kuvaus: {},
      lisatiedot: [],
    },
    julkinen: true,
    esikatselu: true,
    muokkaaja: '1.2.246.562.24.26603962037',
    organisaatioOid: '1.2.246.562.10.60198812368',
    kielivalinta: ['fi', 'sv'],
    ePerusteId: 6777660,
    modified: '2020-09-21T16:27',
  });
};

const getKorkeakouluFields = ({ tyyppi }) => {
  return merge(getBaseFields(), {
    johtaaTutkintoon: true,
    koulutustyyppi: tyyppi,
    koulutuksetKoodiUri: ['koulutus_671112#12'],
    nimi: { fi: 'Fysioterapeutti (AMK)', sv: 'Fysioterapeut (YH)' },
    metadata: {
      tyyppi,
      kuvaus: { fi: 'Fi kuvaus', sv: 'Sv kuvaus' },
      tutkintonimikeKoodiUrit: ['tutkintonimikekk_1#1', 'tutkintonimikekk_2#1'],
      opintojenLaajuusKoodiUri: 'opintojenlaajuus_1#1',
      kuvauksenNimi: { fi: 'Fi kuvauksen nimi', sv: 'Sv kuvauksen nimi' },
      koulutusalaKoodiUrit: [
        'kansallinenkoulutusluokitus2016koulutusalataso2_052#1',
      ],
    },
  });
};

const getLukioFields = ({ tyyppi }) => {
  return merge(getBaseFields({ tyyppi }), {
    koulutuksetKoodiUri: ['koulutus_309902#7'],
    nimi: { fi: 'Lukion oppimäärä' },
    metadata: {
      opintojenLaajuusKoodiUri: 'opintojenlaajuus_150',
    },
  });
};

const getTuvaFields = ({ tyyppi }) => {
  return merge(getBaseFields({ tyyppi }), {
    nimi: { fi: 'Tutkintokoulutukseen valmentava koulutus (TUVA)' },
    metadata: {
      opintojenLaajuusKoodiUri: 'opintojenlaajuus_38',
      kuvaus: { fi: 'kuvausteksti' },
    },
  });
};

const getTelmaFields = ({ tyyppi }) => {
  return merge(getBaseFields({ tyyppi }), {
    nimi: { fi: 'Työhön ja itsenäiseen elämään valmentava koulutus (TELMA)' },
    metadata: {
      opintojenLaajuusKoodiUri: 'opintojenlaajuus_60',
      kuvaus: { fi: 'kuvausteksti' },
    },
  });
};

export default ({ tyyppi = 'amm' } = {}) => {
  if (tyyppi.startsWith('amm')) {
    return getAmmatillinenFields({ tyyppi });
  } else if (['yo', 'amk'].includes(tyyppi)) {
    return getKorkeakouluFields({ tyyppi });
  } else if (tyyppi === 'lk') {
    return getLukioFields({ tyyppi });
  } else if (tyyppi === 'tuva') {
    return getTuvaFields({ tyyppi });
  } else if (tyyppi === 'telma') {
    return getTelmaFields({ tyyppi });
  }

  return getBaseFields();
};
