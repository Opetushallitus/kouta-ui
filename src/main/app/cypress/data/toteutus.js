import merge from 'lodash/merge';

const getBaseFields = () => ({
  oid: '1.2.246.562.17.00000000000000000026',
  koulutusOid: '1.2.246.562.13.00000000000000000041',
  tila: 'tallennettu',
  tarjoajat: ['1.2.246.562.10.74534804344'],
  nimi: { fi: 'Koulutuskeskus Salpaus, jatkuva haku, eläintenhoitaja' },
  metadata: {
    kuvaus: {
      fi: 'Toteutuksen kuvaus',
    },
    opetus: {
      koulutuksenAlkamispaivamaara: '2019-08-23T00:00',
      koulutuksenPaattymispaivamaara: '2019-08-26T00:00',
      opetuskieliKoodiUrit: ['oppilaitoksenopetuskieli_1#1'],
      opetuskieletKuvaus: {
        fi: 'Opetuskieli kuvaus',
      },
      opetusaikaKoodiUrit: ['opetusaikakk_1#1'],
      opetusaikaKuvaus: {
        fi: 'Opetusaika kuvaus',
      },
      opetustapaKoodiUrit: ['opetuspaikkakk_1#1'],
      opetustapaKuvaus: {
        fi: 'Opetustapa kuvaus',
      },
      onkoMaksullinen: true,
      maksullisuusKuvaus: {
        fi: 'Maksullisuus kuvaus',
      },
      maksunMaara: 20,
      alkamiskausiKoodiUri: 'kausi_1#1',
      alkamisvuosi: '2024',
      alkamisaikaKuvaus: {
        fi: 'Alkamisaika kuvaus',
      },
      onkoLukuvuosimaksua: true,
      lukuvuosimaksu: 30,
      lukuvuosimaksuKuvaus: {
        fi: 'Lukuvuosimaksu kuvaus',
      },
      onkoStipendia: true,
      stipendinMaara: {
        fi: '90',
      },
      stipendinKuvaus: {
        fi: 'Stipendin kuvaus',
      },
      lisatiedot: [
        {
          otsikkoKoodiUri: 'koulutuksenlisatiedot_0#1',
          teksti: {
            fi: 'koulutuksenlisatiedot_0 kuvaus',
          },
        },
      ],
    },
    asiasanat: [
      { kieli: 'fi', arvo: 'koira' },
      { kieli: 'fi', arvo: 'eläintenhoito' },
    ],
    ammattinimikkeet: [{ kieli: 'fi', arvo: 'eläintenhoitaja' }],
    yhteyshenkilot: [
      {
        nimi: {
          fi: 'Fi nimi',
          sv: 'Sv nimi',
        },
        puhelinnumero: {
          fi: 'Fi puhelinnumero',
          sv: 'Sv puhelinnumero',
        },
        sahkoposti: {
          fi: 'Fi sähköposti',
          sv: 'Sv sähköposti',
        },
        titteli: {
          fi: 'Fi titteli',
          sv: 'Sv titteli',
        },
        wwwSivu: {
          fi: 'Fi verkkosivu',
          sv: 'Sv verkkosivu',
        },
      },
    ],
  },
  lukionLinjanTarkenneKoodiUrit: ['lukiolinjat2_1#1'],
  erityinenKoulutustehtavaKoodiUrit: ['lukionerityistehtava_1#1'],
  muokkaaja: '1.2.246.562.24.62301161440',
  organisaatioOid: '1.2.246.562.10.594252633210',
  kielivalinta: ['fi', 'sv'],
  modified: '2019-03-26T10:19',
});

const getAmmatillinenFields = ({ tyyppi }) =>
  merge(getBaseFields(), {
    metadata: {
      tyyppi,
      osaamisalat: [
        {
          koodi: 'osaamisala_0',
          linkki: {
            fi: 'https://www.salpaus.fi/koulutusesittely/elaintenhoitaja/',
          },
          otsikko: {
            fi: 'Otsikko',
          },
        },
      ],
    },
  });

const getKorkeakouluFields = ({ tyyppi }) =>
  merge(getBaseFields(), {
    metadata: {
      tyyppi,
      ylemmanKorkeakoulututkinnonOsaamisalat: [
        {
          kuvaus: { fi: 'osaamisalan kuvaus' },
          nimi: { fi: 'osaamisalan nimi' },
          linkki: { fi: 'linkki' },
          otsikko: { fi: 'osaamisalan otsikko' },
        },
      ],
      alemmanKorkeakoulututkinnonOsaamisalat: [
        {
          kuvaus: { fi: 'osaamisalan kuvaus' },
          nimi: { fi: 'osaamisalan nimi' },
          linkki: { fi: 'linkki' },
          otsikko: { fi: 'osaamisalan otsikko' },
        },
      ],
    },
  });

const getLukioFields = ({ tyyppi }) =>
  merge(getBaseFields(), {
    metadata: {
      tyyppi,
      opetus: {
        diplomiKoodiUrit: ['lukiodiplomit_1#1'],
        diplomiKuvaus: {
          fi: 'Fi diplomi',
          sv: 'Sv diplomi',
        },
        A1JaA2Kielivalikoima: ['kieli_1#1'],
        B2Kielivalikoima: ['kieli_2#1'],
        B1Kielivalikoima: ['kieli_3#1'],
        B3Kielivalikoima: ['kieli_4#1'],
        aidinkieliKielivalikoima: ['kieli_4#1'],
        muuKielivalikoima: ['kieli_4#1'],
      },
      lukiolinjaKoodiUri: 'lukiolinjat_0#1',
      jaksonKuvaus: {
        fi: 'Fi jakso',
        sv: 'Sv jakso',
      },
    },
  });

export default ({ tyyppi = 'amm' }) => {
  if (tyyppi === 'amm') {
    return getAmmatillinenFields({ tyyppi });
  } else if (['yo', 'amk'].includes(tyyppi)) {
    return getKorkeakouluFields({ tyyppi });
  } else if (tyyppi === 'lk') {
    return getLukioFields({ tyyppi });
  }

  return getBaseFields();
};
