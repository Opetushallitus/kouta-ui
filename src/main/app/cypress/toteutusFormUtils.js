import merge from 'lodash/merge';

import {
  stubKoodistoRoute,
  stubOppijanumerorekisteriHenkiloRoute,
} from './utils';
import organisaatio from './data/organisaatio';
import organisaatioHierarkia from './data/organisaatioHierarkia';
import koodisto from './data/koodisto';

export const stubToteutusFormRoutes = ({ cy, organisaatioOid, perusteId }) => {
  cy.server();

  cy.route({
    method: 'GET',
    url: `**/organisaatio-service/rest/organisaatio/v4/hierarkia/hae?oid=${organisaatioOid}**`,
    response: organisaatioHierarkia({ rootOid: organisaatioOid }),
  });

  cy.route({
    method: 'GET',
    url: `**/organisaatio-service/rest/organisaatio/v4/${organisaatioOid}**`,
    response: merge(organisaatio(), {
      oid: organisaatioOid,
    }),
  });

  cy.route({
    method: 'POST',
    url: '**/organisaatio-service/rest/organisaatio/v4/findbyoids',
    response: [
      merge(organisaatio(), {
        oid: organisaatioOid,
      }),
    ],
  });

  stubKoodistoRoute({ koodisto: 'koulutuksenlisatiedot', cy });
  stubKoodistoRoute({ koodisto: 'oppilaitoksenopetuskieli', cy });
  stubKoodistoRoute({ koodisto: 'opetusaikakk', cy });
  stubKoodistoRoute({ koodisto: 'opetuspaikkakk', cy });
  stubKoodistoRoute({ koodisto: 'kausi', cy });
  stubKoodistoRoute({ koodisto: 'kieli', cy });
  stubKoodistoRoute({ koodisto: 'lukiodiplomit', cy });
  stubKoodistoRoute({ koodisto: 'lukiolinjat', cy });

  cy.route({
    method: 'GET',
    url:
      '**/koodisto-service/rest/json/relaatio/sisaltyy-ylakoodit/koulutustyyppi_*',
    response: koodisto({ koodisto: 'koulutus' }),
  });

  cy.route({
    method: 'GET',
    url: '**/toteutus/list**',
    response: [],
  });

  cy.route({
    method: 'GET',
    url: '**/koodisto-service/rest/codeelement/koulutus_0',
    response: [
      {
        koodiUri: 'koulutus_0',
        versio: 1,
        metadata: [{ kieli: 'fi', nimi: 'Nimi' }],
      },
    ],
  });

  cy.route({
    method: 'GET',
    url:
      '**/eperusteet-service/api/perusteet?tuleva=true&siirtyma=false&voimassaolo=true&poistunut=false&kieli=fi&koulutuskoodi=koulutus_0',
    response: {
      data: [
        {
          kuvaus: { fi: 'koulutus_0 kuvaus' },
          osaamisalat: [
            { uri: 'osaamisala_0', nimi: { fi: 'osaamisala_0 nimi' } },
          ],
          tutkintonimikeKoodiUri: 'nimike_1#1',
          id: perusteId,
        },
      ],
    },
  });

  cy.route({
    method: 'GET',
    url: `**/eperusteet-service/api/perusteet/${perusteId}/osaamisalakuvaukset`,
    response: {
      data: {
        reformi: {},
      },
    },
  });

  cy.route({
    method: 'GET',
    url:
      '**/koodisto-service/rest/json/relaatio/sisaltyy-alakoodit/koulutus_0**',
    response: [
      {
        koodiUri: 'okmohjauksenala_1',
        metadata: [{ kieli: 'fi', nimi: 'Ala' }],
      },
      {
        koodiUri: 'opintojenlaajuus_1',
        metadata: [{ kieli: 'fi', nimi: '180' }],
      },
      {
        koodiUri: 'opintojenlaajuusyksikko_1',
        metadata: [{ kieli: 'fi', nimi: 'op' }],
      },
    ],
  });

  cy.route({
    method: 'GET',
    url: '**/ammattinimike/search/**',
    response: [],
  });

  cy.route({
    method: 'GET',
    url: '**/asiasana/search/**',
    response: [],
  });

  stubOppijanumerorekisteriHenkiloRoute({ cy });
};
