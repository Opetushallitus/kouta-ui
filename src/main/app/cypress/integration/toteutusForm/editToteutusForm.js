import merge from 'lodash/merge';

import { getByTestId, chooseKieliversiotLanguages } from '../../utils';
import koulutus from '../../data/koulutus';
import toteutus from '../../data/toteutus';
import { stubToteutusFormRoutes } from '../../toteutusFormUtils';

const fillKieliversiotSection = cy => {
  getByTestId('kieliversiotSection', cy).within(() => {
    chooseKieliversiotLanguages(['fi'], cy);
  });
};

const tallenna = cy => {
  getByTestId('tallennaToteutusButton', cy).click();
};

describe('editToteutusForm', () => {
  const organisaatioOid = '1.1.1.1.1.1';
  const koulutusOid = '1.2.1.1.1.1';
  const toteutusOid = '1.3.1.1.1.1';
  const perusteId = '1';

  const testKoulutusFields = {
    oid: koulutusOid,
    organisaatioOid: organisaatioOid,
    koulutusKoodiUri: 'koulutus_0#1',
    tarjoajat: ['4.1.1.1.1.1', '2.1.1.1.1.1'],
  };

  const testToteutusFields = {
    oid: toteutusOid,
    tarjoajat: ['5.1.1.1.1.1', '3.1.1.1.1.1'],
    organisaatioOid: organisaatioOid,
    koulutusOid: koulutusOid,
  };

  beforeEach(() => {
    cy.server();

    stubToteutusFormRoutes({ cy, organisaatioOid, perusteId });

    cy.route({
      method: 'GET',
      url: `**/toteutus/${toteutusOid}/hakukohteet/list**`,
      response: [],
    });

    cy.route({
      method: 'GET',
      url: `**/haku/list**`,
      response: [],
    });

    cy.visit(`/toteutus/${toteutusOid}/muokkaus`);
  });

  it('should be able to edit ammatillinen toteutus', () => {
    cy.route({
      method: 'GET',
      url: `**/koulutus/${koulutusOid}`,
      response: merge(koulutus({ tyyppi: 'amm' }), testKoulutusFields),
    });

    cy.route({
      method: 'GET',
      url: `**/toteutus/${toteutusOid}`,
      response: merge(toteutus({ tyyppi: 'amm' }), testToteutusFields),
    });

    cy.route({
      method: 'POST',
      url: '**/toteutus',
      response: {
        muokattu: false,
      },
    }).as('updateAmmToteutusResponse');

    fillKieliversiotSection(cy);
    tallenna(cy);

    cy.wait('@updateAmmToteutusResponse').then(({ request }) => {
      cy.wrap(request.body).snapshot();
    });
  });

  it('should be able to edit korkeakoulu toteutus', () => {
    cy.route({
      method: 'GET',
      url: `**/koulutus/${koulutusOid}`,
      response: merge(koulutus({ tyyppi: 'yo' }), testKoulutusFields),
    });

    cy.route({
      method: 'GET',
      url: `**/toteutus/${toteutusOid}`,
      response: merge(toteutus({ tyyppi: 'yo' }), testToteutusFields),
    });

    cy.route({
      method: 'POST',
      url: '**/toteutus',
      response: {
        muokattu: false,
      },
    }).as('updateYoToteutusResponse');

    fillKieliversiotSection(cy);
    tallenna(cy);

    cy.wait('@updateYoToteutusResponse').then(({ request }) => {
      cy.wrap(request.body).snapshot();
    });
  });

  it('should be able to edit lukio toteutus', () => {
    cy.route({
      method: 'GET',
      url: `**/koulutus/${koulutusOid}`,
      response: merge(koulutus({ tyyppi: 'lk' }), testKoulutusFields),
    });

    cy.route({
      method: 'GET',
      url: `**/toteutus/${toteutusOid}`,
      response: merge(toteutus({ tyyppi: 'lk' }), testToteutusFields),
    });

    cy.route({
      method: 'POST',
      url: '**/toteutus',
      response: {
        muokattu: false,
      },
    }).as('updateLkToteutusResponse');

    fillKieliversiotSection(cy);
    tallenna(cy);

    cy.wait('@updateLkToteutusResponse').then(({ request }) => {
      cy.wrap(request.body).snapshot();
    });
  });
});
