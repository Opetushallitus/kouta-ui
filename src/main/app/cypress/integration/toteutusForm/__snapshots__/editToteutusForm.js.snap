exports[`editToteutusForm > should be able to edit ammatillinen toteutus #0`] =
{
  "kielivalinta": [
    "fi"
  ],
  "koulutusOid": "1.2.1.1.1.1",
  "metadata": {
    "alemmanKorkeakoulututkinnonOsaamisalat": [],
    "aloituspaikat": null,
    "ammattinimikkeet": [
      {
        "arvo": "eläintenhoitaja",
        "kieli": "fi"
      }
    ],
    "asiasanat": [
      {
        "arvo": "koira",
        "kieli": "fi"
      },
      {
        "arvo": "eläintenhoito",
        "kieli": "fi"
      }
    ],
    "ilmoittautumislinkki": {},
    "kuvaus": {},
    "laajuus": null,
    "laajuusyksikkoKoodiUri": null,
    "lukiolinjaKoodiUri": null,
    "opetus": {
      "A1JaA2Kielivalikoima": [],
      "B1Kielivalikoima": [],
      "B2Kielivalikoima": [],
      "B3Kielivalikoima": [],
      "aidinkieliKielivalikoima": [],
      "diplomiKoodiUrit": [],
      "diplomiKuvaus": {},
      "koulutuksenAlkamiskausi": "kausi_0#1",
      "koulutuksenAlkamispaivamaara": null,
      "koulutuksenAlkamisvuosi": 2020,
      "koulutuksenPaattymispaivamaara": null,
      "koulutuksenTarkkaAlkamisaika": false,
      "lisatiedot": [
        {
          "otsikkoKoodiUri": "koulutuksenlisatiedot_0#1",
          "teksti": {
            "fi": "koulutuksenlisatiedot_0 kuvaus"
          }
        }
      ],
      "maksullisuusKuvaus": {
        "fi": "Maksullisuus kuvaus"
      },
      "maksunMaara": 20,
      "muuKielivalikoima": [],
      "onkoMaksullinen": true,
      "onkoStipendia": true,
      "opetusaikaKoodiUrit": [
        "opetusaikakk_1#1"
      ],
      "opetusaikaKuvaus": {
        "fi": "Opetusaika kuvaus"
      },
      "opetuskieletKuvaus": {
        "fi": "Opetuskieli kuvaus"
      },
      "opetuskieliKoodiUrit": [
        "oppilaitoksenopetuskieli_1#1"
      ],
      "opetustapaKoodiUrit": [
        "opetuspaikkakk_1#1"
      ],
      "opetustapaKuvaus": {
        "fi": "Opetustapa kuvaus"
      },
      "stipendinKuvaus": {
        "fi": "Stipendin kuvaus"
      },
      "stipendinMaara": 90,
      "suunniteltuKestoKuukaudet": 6,
      "suunniteltuKestoKuvaus": {
        "fi": "Fi suunniteltuKestoKuvaus"
      },
      "suunniteltuKestoVuodet": 2
    },
    "osaamisalat": [
      {
        "koodiUri": "osaamisala_0",
        "linkki": {
          "fi": "https://www.salpaus.fi/koulutusesittely/elaintenhoitaja/"
        },
        "otsikko": {
          "fi": "Otsikko"
        }
      }
    ],
    "toteutusjaksot": [],
    "tutkinnonOsat": [],
    "tyyppi": "amm",
    "yhteyshenkilot": [
      {
        "nimi": {
          "fi": "Fi nimi"
        },
        "puhelinnumero": {
          "fi": "Fi puhelinnumero"
        },
        "sahkoposti": {
          "fi": "Fi sähköposti"
        },
        "titteli": {
          "fi": "Fi titteli"
        },
        "wwwSivu": {
          "fi": "Fi verkkosivu"
        }
      }
    ],
    "ylemmanKorkeakoulututkinnonOsaamisalat": []
  },
  "modified": "2019-03-26T10:19",
  "muokkaaja": "1.2.246.562.24.62301161440",
  "nimi": {
    "fi": "Koulutuskeskus Salpaus, jatkuva haku, eläintenhoitaja"
  },
  "oid": "1.3.1.1.1.1",
  "organisaatioOid": "1.1.1.1.1.1",
  "tarjoajat": [
    "5.1.1.1.1.1",
    "3.1.1.1.1.1"
  ],
  "teemakuva": null,
  "tila": "tallennettu"
};

exports[`editToteutusForm > should be able to edit korkeakoulu toteutus #0`] =
{
  "kielivalinta": [
    "fi"
  ],
  "koulutusOid": "1.2.1.1.1.1",
  "metadata": {
    "alemmanKorkeakoulututkinnonOsaamisalat": [
      {
        "kuvaus": {
          "fi": "osaamisalan kuvaus"
        },
        "linkki": {
          "fi": "http://linkki.com"
        },
        "nimi": {
          "fi": "osaamisalan nimi"
        },
        "otsikko": {
          "fi": "osaamisalan otsikko"
        }
      }
    ],
    "aloituspaikat": null,
    "ammattinimikkeet": [
      {
        "arvo": "eläintenhoitaja",
        "kieli": "fi"
      }
    ],
    "asiasanat": [
      {
        "arvo": "koira",
        "kieli": "fi"
      },
      {
        "arvo": "eläintenhoito",
        "kieli": "fi"
      }
    ],
    "ilmoittautumislinkki": {},
    "kuvaus": {},
    "laajuus": null,
    "laajuusyksikkoKoodiUri": null,
    "lukiolinjaKoodiUri": null,
    "opetus": {
      "A1JaA2Kielivalikoima": [],
      "B1Kielivalikoima": [],
      "B2Kielivalikoima": [],
      "B3Kielivalikoima": [],
      "aidinkieliKielivalikoima": [],
      "diplomiKoodiUrit": [],
      "diplomiKuvaus": {},
      "koulutuksenAlkamiskausi": "kausi_0#1",
      "koulutuksenAlkamispaivamaara": null,
      "koulutuksenAlkamisvuosi": 2020,
      "koulutuksenPaattymispaivamaara": null,
      "koulutuksenTarkkaAlkamisaika": false,
      "lisatiedot": [
        {
          "otsikkoKoodiUri": "koulutuksenlisatiedot_0#1",
          "teksti": {
            "fi": "koulutuksenlisatiedot_0 kuvaus"
          }
        }
      ],
      "maksullisuusKuvaus": {
        "fi": "Maksullisuus kuvaus"
      },
      "maksunMaara": 20,
      "muuKielivalikoima": [],
      "onkoMaksullinen": true,
      "onkoStipendia": true,
      "opetusaikaKoodiUrit": [
        "opetusaikakk_1#1"
      ],
      "opetusaikaKuvaus": {
        "fi": "Opetusaika kuvaus"
      },
      "opetuskieletKuvaus": {
        "fi": "Opetuskieli kuvaus"
      },
      "opetuskieliKoodiUrit": [
        "oppilaitoksenopetuskieli_1#1"
      ],
      "opetustapaKoodiUrit": [
        "opetuspaikkakk_1#1"
      ],
      "opetustapaKuvaus": {
        "fi": "Opetustapa kuvaus"
      },
      "stipendinKuvaus": {
        "fi": "Stipendin kuvaus"
      },
      "stipendinMaara": 90,
      "suunniteltuKestoKuukaudet": 6,
      "suunniteltuKestoKuvaus": {
        "fi": "Fi suunniteltuKestoKuvaus"
      },
      "suunniteltuKestoVuodet": 2
    },
    "osaamisalat": [],
    "toteutusjaksot": [],
    "tutkinnonOsat": [],
    "tyyppi": "yo",
    "yhteyshenkilot": [
      {
        "nimi": {
          "fi": "Fi nimi"
        },
        "puhelinnumero": {
          "fi": "Fi puhelinnumero"
        },
        "sahkoposti": {
          "fi": "Fi sähköposti"
        },
        "titteli": {
          "fi": "Fi titteli"
        },
        "wwwSivu": {
          "fi": "Fi verkkosivu"
        }
      }
    ],
    "ylemmanKorkeakoulututkinnonOsaamisalat": [
      {
        "kuvaus": {
          "fi": "osaamisalan kuvaus"
        },
        "linkki": {
          "fi": "http://linkki.com"
        },
        "nimi": {
          "fi": "osaamisalan nimi"
        },
        "otsikko": {
          "fi": "osaamisalan otsikko"
        }
      }
    ]
  },
  "modified": "2019-03-26T10:19",
  "muokkaaja": "1.2.246.562.24.62301161440",
  "nimi": {
    "fi": "Koulutuskeskus Salpaus, jatkuva haku, eläintenhoitaja"
  },
  "oid": "1.3.1.1.1.1",
  "organisaatioOid": "1.1.1.1.1.1",
  "tarjoajat": [
    "5.1.1.1.1.1",
    "3.1.1.1.1.1"
  ],
  "teemakuva": null,
  "tila": "tallennettu"
};

exports[`editToteutusForm > should be able to edit lukio toteutus #0`] =
{
  "kielivalinta": [
    "fi"
  ],
  "koulutusOid": "1.2.1.1.1.1",
  "metadata": {
    "alemmanKorkeakoulututkinnonOsaamisalat": [],
    "aloituspaikat": null,
    "ammattinimikkeet": [
      {
        "arvo": "eläintenhoitaja",
        "kieli": "fi"
      }
    ],
    "asiasanat": [
      {
        "arvo": "koira",
        "kieli": "fi"
      },
      {
        "arvo": "eläintenhoito",
        "kieli": "fi"
      }
    ],
    "ilmoittautumislinkki": {},
    "kuvaus": {},
    "laajuus": null,
    "laajuusyksikkoKoodiUri": null,
    "lukiolinjaKoodiUri": "lukiolinjat_0#1",
    "opetus": {
      "A1JaA2Kielivalikoima": [
        "kieli_1#1"
      ],
      "B1Kielivalikoima": [
        "kieli_3#1"
      ],
      "B2Kielivalikoima": [
        "kieli_2#1"
      ],
      "B3Kielivalikoima": [
        "kieli_4#1"
      ],
      "aidinkieliKielivalikoima": [
        "kieli_4#1"
      ],
      "diplomiKoodiUrit": [
        "lukiodiplomit_1#1"
      ],
      "diplomiKuvaus": {
        "fi": "Fi diplomi"
      },
      "koulutuksenAlkamiskausi": "kausi_0#1",
      "koulutuksenAlkamispaivamaara": null,
      "koulutuksenAlkamisvuosi": 2020,
      "koulutuksenPaattymispaivamaara": null,
      "koulutuksenTarkkaAlkamisaika": false,
      "lisatiedot": [
        {
          "otsikkoKoodiUri": "koulutuksenlisatiedot_0#1",
          "teksti": {
            "fi": "koulutuksenlisatiedot_0 kuvaus"
          }
        }
      ],
      "maksullisuusKuvaus": {
        "fi": "Maksullisuus kuvaus"
      },
      "maksunMaara": 20,
      "muuKielivalikoima": [
        "kieli_4#1"
      ],
      "onkoMaksullinen": true,
      "onkoStipendia": true,
      "opetusaikaKoodiUrit": [
        "opetusaikakk_1#1"
      ],
      "opetusaikaKuvaus": {
        "fi": "Opetusaika kuvaus"
      },
      "opetuskieletKuvaus": {
        "fi": "Opetuskieli kuvaus"
      },
      "opetuskieliKoodiUrit": [
        "oppilaitoksenopetuskieli_1#1"
      ],
      "opetustapaKoodiUrit": [
        "opetuspaikkakk_1#1"
      ],
      "opetustapaKuvaus": {
        "fi": "Opetustapa kuvaus"
      },
      "stipendinKuvaus": {
        "fi": "Stipendin kuvaus"
      },
      "stipendinMaara": 90,
      "suunniteltuKestoKuukaudet": 6,
      "suunniteltuKestoKuvaus": {
        "fi": "Fi suunniteltuKestoKuvaus"
      },
      "suunniteltuKestoVuodet": 2
    },
    "osaamisalat": [],
    "toteutusjaksot": [],
    "tutkinnonOsat": [],
    "tyyppi": "lk",
    "yhteyshenkilot": [
      {
        "nimi": {
          "fi": "Fi nimi"
        },
        "puhelinnumero": {
          "fi": "Fi puhelinnumero"
        },
        "sahkoposti": {
          "fi": "Fi sähköposti"
        },
        "titteli": {
          "fi": "Fi titteli"
        },
        "wwwSivu": {
          "fi": "Fi verkkosivu"
        }
      }
    ],
    "ylemmanKorkeakoulututkinnonOsaamisalat": []
  },
  "modified": "2019-03-26T10:19",
  "muokkaaja": "1.2.246.562.24.62301161440",
  "nimi": {
    "fi": "Koulutuskeskus Salpaus, jatkuva haku, eläintenhoitaja"
  },
  "oid": "1.3.1.1.1.1",
  "organisaatioOid": "1.1.1.1.1.1",
  "tarjoajat": [
    "5.1.1.1.1.1",
    "3.1.1.1.1.1"
  ],
  "teemakuva": null,
  "tila": "tallennettu"
};
